scripts/src/ui/page/multichoice.question.tsx

```diff
@@ -1,29 +1,53 @@
 import * as React from "react";
 import BasePage from "./base";
 import "../../../../styles/src/page/multichoice.question.styl";
+import { ButtonHTMLAttributes } from "react";
 
 export interface FullPageMultiChoiceProps {
 	id: string;
 	title: string;
+	description: string;
 	choices: string[];
-	onSelected?: (answerData: {}) => void;
+	onSelected: (answerData: {}) => void;
 }
 
-class MultichoiceQuestion extends React.Component<any, FullPageMultiChoiceProps> {
-	public render() {
-		const choices = this.props.choices.map((value: string, index: number) => {
-			return <button key={index} className="choice" onClick={() => {
+export interface FullPageMultiChoiceState {
+	disabled: boolean;
+}
+
+class MultichoiceQuestion extends React.Component<FullPageMultiChoiceProps, FullPageMultiChoiceState> {
+	private buttons: JSX.Element[] = [];
+
+	private constructor(props: FullPageMultiChoiceProps) {
+		super(props);
+		this.state = {
+			disabled: false,
+		};
+	}
+
+	private onSelect(value: string) {
+		{
+			if (this.state.disabled) {
+				return;
+			}
+			this.setState({ disabled: true }, () => {
 				this.props.onSelected({ [ this.props.id ]: value });
-			}}>{value}</button>;
+			});
+		}
+	}
+
+	public render() {
+		this.buttons = this.props.choices.map((value: string, index: number) => {
```

Comment:
you probably don't need to annotate the types for the db function, the compiler should infer them: `(value, index) => ...`